clowder-framework/pyclowder
# 70 add support for using new v2 visualization endpoints

Steps for reviewing this PR:

1. Run the Clowder stack (all services, frontend, and backend)

2. Install and run the example image preview extractor (sample-extractors/image-preview-v2) by following the instructions in its README. The Docker instructions won't work now as it uses the PyClowder image as the base. I have changed the requirements.txt to use `git+https://github.com/clowder-framework/pyclowder.git@70-add-support-for-using-new-v2-visualization-endpoints`. It'll be changed to `pyclowder` before merging. 

3. Upload an image file to the Clowder instance, and you should see the updates in the extractor logs. Once Clowder's visualization download URL is updated, the new visualization (image preview) should appear in the user interface.

Note: The following environment variables need to be set when running the extractor:
```
CLOWDER_VERSION=2
IMAGE_BINARY="<Output of `which convert` command>"
```
ImageMagick can be installed from here: https://imagemagick.org/script/download.php
This extractor uses the `convert` tool from ImageMagick. You can find the installed binary file path of `convert` using the `which convert` command and use that in the IMAGE_BINARY environment variable above.
